lib/taiko.js

```diff
@@ -370,31 +398,36 @@ module.exports.client = () => clientProxy;
  * # switch using regex Title
  * await switchTo(/Go*gle/)
  *
- * @param {string} targetUrl - URL/Page title of the tab to switch.
+ * @param {string} arg - URL/Page title of the tab to switch.
  *
  * @returns {Promise}
  */
 
-module.exports.switchTo = async targetUrl => {
+module.exports.switchTo = async arg => {
   validate();
-  if (
-    typeof targetUrl != 'string' &&
-    !Object.prototype.toString.call(targetUrl).includes('RegExp')
-  ) {
-    throw new TypeError(
-      'The "targetUrl" argument must be of type string or regex. Received type ' + typeof targetUrl,
-    );
-  }
-  if (Object.prototype.toString.call(targetUrl).includes('RegExp')) {
-    targetUrl = new RegExp(targetUrl);
-  }
-  const targets = await targetHandler.getCriTargets(targetUrl, currentHost, currentPort);
-  if (targets.matching.length === 0) {
-    throw new Error(`No tab(s) matching ${targetUrl} found`);
+  if (isObject(arg)) {
```

Comment:
`switchTo` will not work if we pass `RegExp` as argument as `isObject(arg)` will be `true` for  a RegExp.